src/redir.c

```diff
@@ -3778,11 +3778,11 @@ int redir_main(struct redir_t *redir,
             setenv("CHI_SESSION_ID", conn.s_state.sessionid, 1);
             setenv("CHI_USERNAME", conn.s_state.redir.username, 1);
             setenv("CHI_USERURL", conn.s_state.redir.userurl, 1);
-            snprintf(buffer, sizeof(buffer), "%lu", conn.s_state.input_octets);
+            snprintf(buffer, sizeof(buffer), "%" PRIu64, conn.s_state.input_octets);
             setenv("CHI_INPUT_BYTES", buffer, 1);
-            snprintf(buffer, sizeof(buffer), "%lu", conn.s_state.output_octets);
+            snprintf(buffer, sizeof(buffer), "%" PRIu64, conn.s_state.output_octets);
             setenv("CHI_OUTPUT_BYTES", buffer, 1);
-            snprintf(buffer, sizeof(buffer), "%lu", conn.s_params.sessiontimeout);
+            snprintf(buffer, sizeof(buffer), "%" PRIu64, conn.s_params.sessiontimeout);
             setenv("CHI_SESSION_TIMEOUT", buffer, 1);
```

Comment:
You should look at what `time_t` is set to in `sys/types.h` on your platform, these changes do not work on a system with 64bit `time_t` which is a `uint64_t`
`redir.c:3781: warning: format '%lu' expects type 'long unsigned int', but argument 4 has type 'uint64_t'`
`redir.c:3783: warning: format '%lu' expects type 'long unsigned int', but argument 4 has type 'uint64_t'`
`redir.c:3785: warning: format '%lu' expects type 'long unsigned int', but argument 4 has type 'uint64_t'`